graphql/lessonplan/mutations.ts

```diff
@@ -0,0 +1,219 @@
+import { Resolver, Mutation, Arg, Authorized, Ctx, InputType, Field, Int, ObjectType, registerEnumType } from 'type-graphql';
+import { MaxLength, ArrayMaxSize, validateOrReject } from 'class-validator';
+import { Role } from '../authorizations';
+import { GraphQLContext } from '../context';
+import { getLogger } from '../../common/logger/logger';
+import { generateLessonPlan } from '../../common/lessonplan/generate';
+import { course_subject_enum, pupil_state_enum, school_schooltype_enum } from '@prisma/client';
+import { ApolloError, UserInputError } from 'apollo-server-express';
+
+const logger = getLogger(`LessonPlan Mutations`);
+
+// Maximum prompt length (in characters)
+const MAX_PROMPT_LENGTH = 10000;
+
+// Maximum number of files
+const MAX_FILES = 10;
+
+// Register the course_subject_enum as a GraphQL enum
+registerEnumType(course_subject_enum, {
+    name: 'CourseSubjectEnum',
+    description: 'The subject of the course',
+});
+
+// Register the school_schooltype_enum as a GraphQL enum
+registerEnumType(school_schooltype_enum, {
+    name: 'SchoolSchoolTypeEnum',
+    description: 'The type of school',
+});
+
+// Create a new enum for expected output fields
+enum LessonPlanField {
+    TITLE = 'title',
+    LEARNING_GOAL = 'learningGoal',
+    AGENDA_EXERCISES = 'agendaExercises',
+    ASSESSMENT = 'assessment',
+    HOMEWORK = 'homework',
+    RESOURCES = 'resources',
+}
+
+// Register the LessonPlanField enum
+registerEnumType(LessonPlanField, {
+    name: 'LessonPlanField',
+    description: 'Fields that can be generated for a lesson plan',
+});
+
+@InputType()
+class GenerateLessonPlanInput {
+    @Field(() => [String], { nullable: true })
+    @ArrayMaxSize(MAX_FILES, { message: `Number of files must not exceed ${MAX_FILES}` })
+    fileUuids?: string[];
+
+    @Field(() => course_subject_enum)
+    subject: course_subject_enum;
+
+    @Field(() => Int)
+    grade: number;
+
+    @Field(() => Int)
+    duration: number;
+
+    @Field(() => pupil_state_enum)
+    state: pupil_state_enum;
```

Comment:
state, grade, schoolType come from the pupil, so we could just pass in the matchId for which the lessonplan is generated (might make it easier to extend the prompt in the future)?